lib/content-api-client.js

```diff
@@ -0,0 +1,99 @@
+require('colors');
+const NetworkUtils = require('./utils/NetworkUtils');
+const {
+    renderedRegionsByPageTypeQuery,
+    renderedRegionsByPageTypeAndEntityIdQuery,
+} = require('./graphql/query');
+
+const networkUtils = new NetworkUtils();
+
+/**
+ * @param {object} options
+ * @param {string} options.accessToken
+ * @param {string} options.storeUrl
+ * @param {string} pageType
+ * @returns {Promise<renderedRegions: array}>}
+ */
+async function getRenderedRegionsByPageType({ accessToken, storeUrl, pageType }) {
+    try {
+        const query = renderedRegionsByPageTypeQuery(pageType);
+
+        const response = await networkUtils.sendApiRequest({
+            url: `${storeUrl}/graphql`,
+            headers: {
+                'cache-control': 'no-cache',
+                'content-type': 'application/json',
+                Authorization: `Bearer ${accessToken}`,
+            },
+            method: 'POST',
+            data: JSON.stringify({
+                query,
+            }),
+        });
+
+        const {
+            site: { content },
+        } = response.data.data;
+        const {
+            renderedRegionsByPageType: { regions },
+        } = content;
+
+        return {
+            renderedRegions: regions,
+        };
+    } catch (err) {
+        console.log(err);
+        throw err;
+    }
+}
+
+/**
+ * @param {object} options
+ * @param {string} options.accessToken
+ * @param {string} options.storeUrl
+ * @param {string} pageType
+ * @param {number} entityId
+ * @returns {Promise<renderedRegions: array}>}
+ */
+async function getRenderedRegionsByPageTypeAndEntityId({
```

Comment:
Kinda surprised these need to be two separate functions, isn't really just affecting the query slightly when you don't supply an entity id?

Reply:
Oh wait right as I clicked the button I realized that parsing the response is slightly different too, yeah fair enough.

Reply:
When using the GraphQL Playground to build the query it pretty handy because it does the validation of whether a pageType needs entityId or not. I am guessing that was the reasoning for having separate function as its difficult to make a argument conditionally required  on the same function.

Reply:
Oh sorry I was speaking of *these* functions, not the graphql query itself. But anyway yeah given that it's a different graphql field think this makes sense.